pkg/upload/js.go

```diff
@@ -146,7 +242,22 @@ func uploadSingleSourceMap(
 	logger log.Logger,
 ) {
 
-	bundlePath, err := resolveBundlePath(bundlePath, sourceMapPath)
+	sourceMapContents, err := ReadSourceMap(sourceMapPath, logger)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	AddSources(sourceMapContents, projectRoot, logger)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	encodedSourceMap, err := json.Marshal(sourceMapContents)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
```

Comment:
I don't know what the likleihood is of an error unmarshalling the JSON is, but I think I'd feel happier if we used the original content if we detect the `sourcesContent` is already present; and also if there was an error in the un/marshalling itself. So if it was an unexpected file format, we just continue to upload with a warning that no sources were attached.

I'm open to other views on this though. @imjoehaines/@lemnik - any thoughts here on the most desirable behaviour?